apps/src/music/views/ChordPanel.tsx

```diff
@@ -0,0 +1,222 @@
+import React, {useState, useEffect, useRef} from 'react';
+import {getNoteName, isBlackKey} from '../utils/Notes';
+import MusicLibrary from '../player/MusicLibrary';
+import {ChordEventValue, PlayStyle} from '../player/interfaces/ChordEvent';
+
+const FontAwesome = require('../../templates/FontAwesome');
+const moduleStyles = require('./chordPanel.module.scss').default;
+const classNames = require('classnames');
+
+const NUM_OCTAVES = 3;
+const START_OCTAVE = 4;
+const MAX_NOTES = 16;
+
+const styleDropdownOptions: [PlayStyle, string][] = [
+  ['arpeggio-up', 'Arpeggio Up'],
+  ['arpeggio-down', 'Arpeggio Down'],
+  ['arpeggio-random', 'Arpeggio Random'],
+  ['together', 'Together']
+];
+
+export interface ChordPanelProps {
+  library: MusicLibrary;
+  initValue: ChordEventValue;
+  onChange: (value: ChordEventValue) => void;
+  previewChord: (chord: ChordEventValue, onStop: () => void) => void;
+}
+
+const ChordPanel: React.FunctionComponent<ChordPanelProps> = ({
+  initValue,
+  onChange,
+  previewChord,
+  library
+}) => {
+  const [selectedNotes, setSelectedNotes] = useState<number[]>(initValue.notes);
+  const [playStyle, setPlayStyle] = useState<PlayStyle>(initValue.playStyle);
+  const [instrument, setInstrument] = useState<string>(initValue.instrument);
+  const [isPlayingPreview, setIsPlayingPreview] = useState<boolean>(false);
+  const [isDisabled, setIsDisabled] = useState<boolean>(
+    selectedNotes.length >= MAX_NOTES
+  );
+
+  const instruments: [string, string][] = library.groups[0].folders
+    .filter(folder => folder.type === 'instrument')
+    .map(folder => [folder.name, folder.path]);
+
+  const onPressKey = (note: number) => {
```

Comment:
I saw `onPressKey` and went to see if there were keyboard shortcuts, but now it makes sense. :)

It does feel like hearing a preview of a key as it's toggled on could be quite helpful, but we can refine the experience later of course.

Reply:
Agreed! I'll add that as a follow-up.

Reply:
https://codedotorg.atlassian.net/browse/SL-684